R/deps.R

```diff
@@ -6,15 +6,28 @@ idesc_set_dep <- function(self, private, package, type, version) {
 
   if (length(has)) {
     deps[ has, "version" ] <- version
-
   } else {
-    deps <- rbind(
-      deps,
-      data.frame(
-        stringsAsFactors = FALSE,
-        type = type, package = package, version = version
-      )
+    row <- data.frame(
+      stringsAsFactors = FALSE,
+      type = type, package = package, version = version
     )
+    others <- deps$package[deps$type == type]
+    sorted <- !is.unsorted(others) && length(others) > 0
+
+    if (sorted) {
+      # find first row it should come after
```

Comment:
Can't we just add it to the bottom, and then sort the df, if it was already sorted?

Reply:
But you have to consider the sorting within a section (i.e. the imports might be sorted but the suggests is not). I tried a few approaches and this seemed the most robust to me (despite being rather complicated)

Reply:
Right. I suspected that it is not so simple.